sql_export/sql.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2015 Akretion (<http://www.akretion.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv.orm import Model
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+import StringIO
+import base64
+import time
+
+
+PROHIBITED_WORDS = [
+    'delete',
+    'drop',
+    'insert',
+    'alter',
+    'truncate',
+    'execute',
+]
+
+
+class SqlExport(Model):
+    _name = "sql.export"
+    _description = "SQL export"
+
+    def _check_query_allowed(self, cr, uid, ids, context=None):
+        for obj in self.browse(cr, uid, ids, context=context):
+            if any(word in obj.query.lower() for word in PROHIBITED_WORDS):
+                return False
+            if obj.query.strip().split(' ')[0].lower() != 'select':
```

Comment:
IMO the second `if` is too restrictive.  It's sometimes useful to build query using the `with` clause (http://www.postgresql.org/docs/9.1/static/queries-with.html). Moreover, if you want to test that the query starts with a select you should write
```python
            if not obj.query.lower().startswith('select'):
                return False
```
IMO, checking for prohibited works is enough.